helix-term/src/ui/editor.rs

```diff
@@ -375,12 +398,25 @@ impl EditorView {
                             }
                         } else {
                             let grapheme = Cow::from(grapheme);
+                            let is_whitespace;
 
                             let (grapheme, width) = if grapheme == "\t" {
+                                is_whitespace = true;
                                 // make sure we display tab as appropriate amount of spaces
                                 let visual_tab_width = tab_width - (visual_x as usize % tab_width);
-                                (&tab[..visual_tab_width], visual_tab_width)
+                                let grapheme_tab_width = tab
+                                    .grapheme_indices(true)
+                                    .nth(visual_tab_width)
+                                    .map(|(i, _)| i)
+                                    .unwrap_or(tab.len());
+
+                                (&tab[..grapheme_tab_width], visual_tab_width)
```

Comment:
Is this too hacky? I was trying to get an iterator over the graphemes and `.take(visual_tab_width)` but I ran into borrowing troubles